include/xsimd/types/xsimd_neon_uint16.hpp

```diff
@@ -0,0 +1,390 @@
+    }
+
+    inline batch<uint16_t, 8>& batch<uint16_t, 8>::load_unaligned(const int16_t* src)
+    {
+        return load_aligned(src);
+    }
+
+
+    inline batch<uint16_t, 8>& batch<uint16_t, 8>::load_aligned(const uint16_t* src)
+    {
+        m_value = vld1q_u16(src);
+        return *this;
+    }
+
+    inline batch<uint16_t, 8>& batch<uint16_t, 8>::load_unaligned(const uint16_t* src)
+    {
+        return load_aligned(src);
+    }
+
+    inline void batch<uint16_t, 8>::store_aligned(int16_t* dst) const
+    {
+        vst1q_s16(dst, m_value);
+    }
+
+    inline void batch<uint16_t, 8>::store_unaligned(int16_t* dst) const
+    {
+        store_aligned(dst);
+    }
+
+    inline void batch<uint16_t, 8>::store_aligned(uint16_t* dst) const
+    {
+        vst1q_u16(dst, m_value);
+    }
+
+    inline void batch<uint16_t, 8>::store_unaligned(uint16_t* dst) const
+    {
+        store_aligned(dst);
+    }
+
+    XSIMD_DEFINE_LOAD_STORE_INT16(uint16_t, 8, 8)
+
+    inline batch<uint16_t, 8>::operator uint16x8_t() const
+    {
+        return m_value;
+    }
+
+    inline uint16_t batch<uint16_t, 8>::operator[](std::size_t index) const
+    {
+        return m_value[index];
+    }
+
+    namespace detail
+    {
+
+#define XSIMD_INLINE_UNROLL_OP(op, i)  \
+    static_cast<uint16_t>(lhs[i] op rhs[i])
+
+#define XSIMD_INLINE_UNROLL_OP_8(op)  \
+    XSIMD_INLINE_UNROLL_OP(op, 0),     \
+    XSIMD_INLINE_UNROLL_OP(op, 1),     \
+    XSIMD_INLINE_UNROLL_OP(op, 2),     \
+    XSIMD_INLINE_UNROLL_OP(op, 3),     \
+    XSIMD_INLINE_UNROLL_OP(op, 4),     \
+    XSIMD_INLINE_UNROLL_OP(op, 5),     \
+    XSIMD_INLINE_UNROLL_OP(op, 6),     \
+    XSIMD_INLINE_UNROLL_OP(op, 7),
+
+        template <>
+        struct batch_kernel<uint16_t, 8>
+        {
+            using batch_type = batch<uint16_t, 8>;
+            using value_type = uint16_t;
+            using batch_bool_type = batch_bool<uint16_t, 8>;
+
+            static batch_type neg(const batch_type& rhs)
+            {
+                return vreinterpretq_u16_s16(vnegq_s16(vreinterpretq_s16_u16(rhs)));
+            }
+
+            static batch_type add(const batch_type& lhs, const batch_type& rhs)
+            {
+                return vaddq_u16(lhs, rhs);
+            }
+
+            static batch_type sub(const batch_type& lhs, const batch_type& rhs)
+            {
+                return vsubq_u16(lhs, rhs);
+            }
+
+            static batch_type mul(const batch_type& lhs, const batch_type& rhs)
+            {
+                return vmulq_u16(lhs, rhs);
+            }
+
+            static batch_type div(const batch_type& lhs, const batch_type& rhs)
+            {
+                return uint16x8_t{
+                    XSIMD_INLINE_UNROLL_OP_8(/)
+                };
+            }
+
+            static batch_type mod(const batch_type& lhs, const batch_type& rhs)
+            {
+                return uint16x8_t{
+                    XSIMD_INLINE_UNROLL_OP_8(%)
```

Comment:
maybe not! :)

Reply:
It is a bit confusing, but `XSIMD_INLINE_UNROLL_OP_8` is for 8 elements, not for int8. And we have 8 elements of uint16 in the batch here.

Reply:
Or maybe I don't understand the problem you're seeing?

Reply:
Maybe we should rename the macro `XSIMD_INLINE_UNROLL_OP_UINT16x8` so that it's not confusing anymore

Reply:
Sorry, I wasn't clear. I was referring to my statement above regarding usage of the meta-unroller. 
But I think we can't use it in a initializer, so a MACRO is necessary.

Reply:
I made a template unroller like you suggested :) better than the macro

Reply:
looks great!